OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1467 @@
+                }
+                val lifecycleList = l.toList
+                makeEdge(lifecycleList.head, lifecycleList(1))
+                makeEdge(lifecycleList.head, lifecycleList(2))
+                makeEdge(lifecycleList(1), lifecycleList(4))
+                makeEdge(lifecycleList(2), lifecycleList(3))
+                makeEdge(lifecycleList(3), lifecycleList(4))
+                makeEdge(lifecycleList(1), lifecycleList(4))
+            } else {
+                makeEdge(create, startComm)
+                makeEdge(create, bind)
+                makeEdge(startComm, destroy)
+                makeEdge(bind, unbind)
+                makeEdge(unbind, destroy)
+                makeEdge(create, destroy)
+            }
+        }
+    }
+
+    def makeEdge(
+        org: Method,
+        rec: Method
+    ): Unit = {
+        edges += ((org, 0) -> rec)
+    }
+
+    def makeEdge(
+        org: Option[Method],
+        rec: Option[Method]
+    ): Unit = {
+        if (org.isDefined && rec.isDefined) {
+            edges += ((org.get, 0) -> rec.get)
+        }
+    }
+
+    /**
+     * Searches all Intents and callbacks in the project.
+     */
+    def searchIntentsAndCallbacks(): ProperPropertyComputationResult = {
+        val dec = project.get(DeclaredMethodsKey)
+        val partialResults = ListBuffer.empty[ProperPropertyComputationResult]
+        project.allMethodsWithBody.filter(dec(_) != null).foreach(m ⇒ {
+            val tacaiEP = propertyStore(m, TACAI.key)
+            if (tacaiEP.hasUBP && tacaiEP.ub.tac.isDefined) {
+                processMethod(m)
+            } else {
+                leftOverMethods += m
+                partialResults += InterimPartialResult(Nil, Set(tacaiEP), continuationForTAC(m))
+            }
+        })
+        if (leftOverMethods.isEmpty) {
+            startSecondPhase(project)
+        } else Results(partialResults)
+    }
+
+    private[this] def continuationForTAC(m: Method)(eps: SomeEPS): PropertyComputationResult = {
+        eps match {
+            case UBP(tac: TACAI) if tac.tac.isDefined ⇒
+                processMethod(m)
+                leftOverMethods -= m
+                if (leftOverMethods.isEmpty && notStarted) {
+                    //if all methods are processed and the second phase has not started yet, the second phase is started.
+                    notStarted = false
+                    startSecondPhase(project)
+                } else NoResult
+            case _ ⇒
+                InterimPartialResult(
+                    Nil,
+                    Set(eps),
+                    continuationForTAC(m)
+                )
+        }
+    }
+
+    /**
+     * Collects information of intents, intent filters and callbacks from the given method m.
+     * @param m The processed method.
+     */
+    private def processMethod(m: Method): Unit = {
+        val body = m.body.get
+        val intentUseSites: mutable.Map[UByte, List[UByte]] = mutable.Map.empty[UByte, List[UByte]]
```

Comment:
`UByte` is almost certainly the wrong type here. IntelliJ sometimes messes this up, so maybe that was the issue here, but use sites should probably be `Int`. Also, as everywhere, please avoid the use of mutable data structures.